src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/DeviceCode/AcquireTokenByDeviceCodeHandler.cs

```diff
@@ -47,7 +47,7 @@ public AcquireTokenByDeviceCodeHandler(RequestData requestData, DeviceCodeResult
         {
             TimeSpan timeRemaining = deviceCodeResult.ExpiresOn - DateTimeOffset.UtcNow;
             AuthenticationResultEx resultEx = null;
-            while (timeRemaining.TotalSeconds > 0)
+            while (timeRemaining.TotalSeconds + deviceCodeResult.Interval > 0)
```

Comment:
The loop ended before the final request was sent resulting in a null resultEx. Now one more request will be sent to get the final error message from the STS.